tasks/gather_facts_ec2_security_groups.yml

```diff
@@ -0,0 +1,29 @@
+---
+
+###
+### Gather ec2 security facts
+###
+
+- name: "({{ cluster.name }}) set ec2 security filter"
+  set_fact:
+    kops_ec2_security_groups_filters:
+      - key: vpc-id
+        val: "{{ kops_vpc_id }}"
+
+- name: Get Security Group ID
+  ec2_group_facts:
+    aws_access_key: "{{ lookup('ENV', 'AWS_ACCESS_KEY') | default(omit) }}"
+    aws_secret_key: "{{ lookup('ENV', 'AWS_SECRET_KEY') | default(omit)  }}"
+    security_token: "{{ lookup('ENV', 'AWS_SECURITY_TOKEN') | default(omit) }}"
+    profile: "{{ kops_aws_profile | default(omit) }}"
+    filters: "{{ kops_ec2_security_groups_filters | get_attr('key', 'val') }}"
+    region: "{{ cluster.region | default(kops_default_region) }}"
+  register: _kops_api_additional_sgs
+
+- name: Set Security Groups ID fact
+  set_fact:
+    kops_api_additional_sgs: "{{ kops_api_additional_sgs |
+      combine( { item.group_name: item.group_id } ) }}"
```

Comment:
The `combine` filter would just combine specified security groups with all found security groups in a VPC. Is this intended?

Reply:
Yes, this will create a structure like this:
{
 security_grup1: security_group_id,
 security_grup2: security_group_id,
 security_grup3: security_group_id
}

Then at the time to render the template it can be accessed by the name of security groups we specified in the cluster-internet.yml